extra-cpp-bindings/src/lib.rs

```diff
@@ -55,14 +73,74 @@ mod ffi {
         pub size: u32,
     }
 
-    /// wallet connect cronos(eth) legacy-tx signing info
-    pub struct WalletConnectTxLegacy {
+    /// wallet connect cronos(eth) eip155-tx signing info
+    pub struct WalletConnectTxEip155 {
         pub to: String,        // hexstring, "0x..."
         pub gas: String,       // decimal string, "1"
         pub gas_price: String, // decimal string
         pub value: String,     // decimal string, in wei units
         pub data: Vec<u8>,     // data, as bytes
         pub nonce: String,     // decimal string
+        pub chainid: u64,      // integer u64
+    }
+
+    /// wallet connect cronos(eth) erc721-tx signing info
+    pub struct WalletConnectErc721Transfer {
+        pub contract_address: String, // hexstring, "0x..."
+        pub from_address: String,     // hexstring, "0x..."
+        pub to_address: String,       // hexstring, "0x..."
+        pub token_id: String,         // decimal string, "1"
+        pub nonce: String,            // decimal string, "1"
+        pub gas_price: String,        // decimal string, "1"
+        pub gas_limit: String,        // decimal string, "1"
+        pub chain_id: u64,            // integer u64
+        pub web3api_url: String,      // string
+    }
+
+    pub struct WalletConnectErc1155Transfer {
+        pub contract_address: String,
+        pub from_address: String,
+        pub to_address: String,
+        pub token_id: String,
+        pub amount: String,
+        pub additional_data: Vec<u8>,
+        pub nonce: String,
+        pub gas_price: String,
+        pub gas_limit: String,
+        pub chain_id: u64,
+        pub web3api_url: String,
+    }
+
+    pub struct WalletConnectErc1155Batch {
+        pub contract_address: String,
+        pub from_address: String,
+        pub to_address: String,
+        pub token_ids: Vec<String>,
+        pub amounts: Vec<String>,
+        pub additional_data: Vec<u8>,
+        pub nonce: String,
+        pub chain_id: u64,
+        pub web3api_url: String,
+    }
+
+    pub struct WalletConnectErc721Approve {
+        pub contract_address: String,
+        pub from_address: String,
+        pub approved_address: String,
+        pub token_id: String,
+        pub nonce: String,
+        pub chain_id: u64,
+        pub web3api_url: String,
+    }
+
+    pub struct WalletConnectErc1155Approve {
+        pub contract_address: String,
+        pub from_address: String,
+        pub approved_address: String,
+        pub approved: bool,
+        pub nonce: String,
+        pub chain_id: u64,
+        pub web3api_url: String,
```

Comment:
these contract related structs have a lot of duplicate information... perhaps some of it can be extracted to common structs? (If these contract structs are also used for json, those common structs could be "flattened": https://serde.rs/attr-flatten.html so that it's still compatible?)

Reply:
ok, i'll try to use common struct

Reply:
added common struct